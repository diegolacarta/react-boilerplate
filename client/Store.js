import {observable} from 'mobx'

export default class Store {
  @observable data = {}

  fetchData = () => {
    return fetch('/api/data')
      .then(response => response.json())
      .then((data) => {
        this.data = data
      })
  }
}