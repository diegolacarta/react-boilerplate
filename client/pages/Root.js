import React from 'react'

export default class extends React.Component {
  static propTypes = {
    children: React.PropTypes.node.isRequired
  }

  render() {
    return (
      <div>
        <h1>Hello</h1>
        {this.props.children}
      </div>
    )
  }
}