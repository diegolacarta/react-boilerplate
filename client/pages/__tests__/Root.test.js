import React from 'react'
import renderer from 'react-test-renderer'
import Root from '../Root'
import Store from '../../Store'

test('Root', () => {
  const tree = renderer.create(
    <Root store={new Store()}>Hi</Root>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})