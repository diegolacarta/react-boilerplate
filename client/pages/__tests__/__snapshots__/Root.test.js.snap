// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root 1`] = `
<div>
  <h1>
    Hello
  </h1>
  Hi
</div>
`;
